pallets/validators-manager/src/benchmarking.rs

```diff
@@ -10,102 +10,129 @@ use super::*;
+        };
+
+        avn_validators.push(Validator::new(account.clone(), key));
+        validators.push((account, eth_key));
+    }
+
+    // Setup validators in avn pallet
+    let mut new_avn_validators = avn::Validators::<T>::get();
+    new_avn_validators.append(&mut avn_validators);
+    avn::Validators::<T>::put(new_avn_validators);
 
-    return validators
+    validators.iter().enumerate().for_each(|(i, (account_id, eth_public_key))| {
+        force_add_collator::<T>(&account_id, i as u64, &eth_public_key)
+    });
 }
 
-fn setup_action_voting<T: Config>(
-    validators: Vec<Validator<<T as pallet_avn::Config>::AuthorityId, T::AccountId>>,
-) -> (
+fn setup_action_voting<T: Config>() -> (
     Validator<T::AuthorityId, T::AccountId>,
     ActionId<T::AccountId>,
     ecdsa::Signature,
     <T::AuthorityId as RuntimeAppPublic>::Signature,
     u32,
 ) {
-    let sender_index = validators.len() - (1 as usize);
-    let sender: Validator<T::AuthorityId, T::AccountId> = validators[sender_index].clone();
-    let action_account_id: T::AccountId = validators[1].account_id.clone();
-    let ingress_counter: IngressCounter = 1;
-    let action_id: ActionId<T::AccountId> = ActionId::new(action_account_id, ingress_counter);
-    let approval_signature: ecdsa::Signature = ecdsa::Signature::from_slice(&hex!("2b01699be62c1aabaf0dd85f956567ac495d4293323ee1eb79d827d705ff86c80bdd4a26af6f50544af9510e0c21082b94ecb8a8d48d74ee4ebda6605a96d77901")).unwrap().into();
-    let signature: <T::AuthorityId as RuntimeAppPublic>::Signature = generate_signature::<T>();
-    let quorum = setup_voting_session::<T>(&action_id);
+    let (vote_sender_id, vote_sender_key, _) = generate_sender_collator_account_details::<T>();
+    let (action_account_id, _, _) = generate_resigning_collator_account_details::<T>();
 
-    let eth_public_key = Public::from_raw(
-        <[u8; 33]>::from_hex("032717041a29cee520bb901c406e2399974d4ad58cd3ee830848bd78fa089f1335")
-            .unwrap(),
-    );
+    let ingress_counter: IngressCounter = 1;
 
-    EthereumPublicKeys::<T>::insert(eth_public_key.clone(), sender.account_id.clone());
+    let action_id: ActionId<T::AccountId> =
+        ActionId::new(action_account_id.clone(), ingress_counter);
 
-    setup_action_data::<T>(
-        sender.account_id.clone(),
-        action_id.action_account_id.clone(),
-        action_id.ingress_counter,
-    );
+    let signature: <T::AuthorityId as RuntimeAppPublic>::Signature = generate_signature::<T>();
+    let quorum = setup_voting_session::<T>(&action_id);
+    // signed by sender private key [7u8; 32]
+    let approval_signature: ecdsa::Signature = ecdsa::Signature::from_slice(&hex!("c951051534f26f2101224ed0ab56ddaeffbb57952e06166c5033fa2290d8306f046628ede2a45cff90391d0a604f08989ced0177c63febbe3afd5340c38561091c")).unwrap().into();
 
-    (sender, action_id, approval_signature, signature, quorum)
+    setup_resignation_action_data::<T>(vote_sender_id.clone(), action_id.ingress_counter);
```

Comment:
I'm not sure about this. Could we have the test be clearer and make sure our intentions are correct?

If I read correctly, we have a variable action_id as defined by the `generate_xxx_account_details` function(s). But we create a signature for a fixed value of `0`. We also have an ingress counter of `1`. So, it's a mystery to me how these values are related, and I think some renaming can explain that.

So, what is this signature confirming? Why is it fixed at 0? Does it depend on the value of action_id we have in here?

I think it would be good for the `signature` variable to reflect its intentions, for example, `signature_for_current_action_id`.

Reply:
And why/how do we know the signer is private key [7u8; 32]? From what I see, it should be `T::AuthorityId::generate_pair(None);`. Is this fixed? Why is this account signing and where do we verify the signature? How do we ensure it is verified by the same account?

Reply:
I think this goes beyond fixing the benchmarks and probably a general refactor to make this area easier to the eye.
An action though is defined by a validator and an ingress, that action will need some data, that will be identified from that key.
The action can be collator activation or deregistration, in this case, we setup a deregistration.
As for the `signature` variable, this is the unsigned extrinsic signature we attach, probably we don't need to validate it during benchmarks, hence the "dummy" values with 0.